README.md

```diff
@@ -1,99 +1,143 @@
 
-end
+```ruby
+RSpec.describe Person, type: :model do
 ```
 
-This first line is just a Ruby method that takes the name of the class you will be testing, and a second argument of a block of code (do/end).
+This is a [Example Group definition](https://relishapp.com/rspec/rspec-core/v/3-7/docs/example-groups/basic-structure-describe-it). In RSpec our tests are called examples, and we group them together using an example group. The top most example group definition gets prefixed with `RSpec`. Here we are defining a example group that describes the `Person` object as the subject. You will often also see a descriptive string used as the value for the subject. We are passing [meta data](https://relishapp.com/rspec/rspec-core/v/3-7/docs/metadata) to the example group here. This lets provide extra configuration for this example group. The meta data is not required, and addition meta data could be assigned. We pass a block to the example group. This block will contain examples and or more example groups. RSpec provides a `context` method that lets us do the same thing as `descirbe`. `context is an alias for `describe`.
+
+```ruby
+...
 
-## Describe and Context
+describe "#first_name" do
 
-Rspec provides a great way to allow your test code to serve as documentation. The `describe` and `context` methods help by taking 2 arguments. First argument is a string that should identify what you are testing. And the second is a block that contains tests, or more descriptions that contain tests.
+...
+```
 
-These methods do not trigger a test themselves. Instead, they give form to the test file. Providing a layout for what our tests do.
+Here we see a describe with a string. This creates an example group for the `first_name` instance method on the Person object. It is convention to describe instance methods by prefixing the method name with a `#`. To describe a class method convention dictates prefixing with a `.`. There will be times when you are creating an example group that does not describe a method. In those causes the subject will be a descriptive string.
 
-`describe` and `context` perform the exact same way. So why have two methods that do the same thing? Well, our tests are semantic, or "relating to meaning in language". So how we may use the words "context" and "describe" in real life will influence how we group our tests.
+# Examples
 
 ```ruby
-require "spec_helper"
+...
 
-Rspec.describe Tire do
-  describe "repels liquid" do
-    context "on short trips" do
-      # some tests
-    end
+it "is the first name for the person" do
 
-    context "on long trip" do
-      # some tests
-    end
+...
+```
 
-    context "in rain storms" do
-      # some tests
-    end
-  end
+Tests in RSpec are called examples. They are declared with an `it` method. Here we are passing in a string `"is the first name for the person"` to describe what this example is expecting and verifying. The `it` method takes a block, this will be executed with the block runs.
```

Comment:
> The it method takes a block, this will be executed with the block runs.

Seems like this sentance was reworked at some point and the second half seems incomplete. I think what this is trying to say is that the block defines the code which is executed when RSpec 'runs the test' and that the block's code should implement / exercise the expectation stated in the example's description.